### DISCUSSION TITLE
Problems when using `vcpkg_build_make.cmake`

### DISCUSSION THREAD

**Discussion by cpp977** [microsoft/vcpkg#30857] [Q&A] [answered]
I need to build a package by using linux `configure` and `make && make install` in a custom port file.
This port file is then consumed in manifest mode.
For `configure` I use `vcpkg_configure_make.cmake` and it runs without problems.
For the build step with `make` I use `vcpkg_build_make.cmake` and this is also causing no issues.
For the install step, I use `vcpkg_install_make.cmake` which forwards to `vcpkg_build_make.cmake` with the appropriate parameters.
The installation step also runs fine but leads to an error at the end:
```
CMake Error at scripts/cmake/vcpkg_build_make.cmake:192 (file):
  file RENAME failed to rename

    /path/to/vcpkg/packages/<package_name>_<triplet>_tmp/path/to/build/vcpkg_installed/x64-linux

  to

    /path/to/vcpkg/packages/ctf_x64-linux
```
I could not figure out the purpose of this rename and the purpose of the whole if block at this position in the file `vcpkg_build_make.cmake` .
If I comment out the rename statement, it works without errors.
The only issue which I am facing then is if the package get installed from the cache (without rebuild): Then nothing happens and the library and header files are not installed into the appropriate directories. (They get properly installed if the cache is not used). Maybe this is related to the above error?
Any help is highly appreciated.

Edit: I found a related issue: #11874

The custom port is as follows:
```
vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO some_repo
    REF some_ref
    SHA512 some_sha
    HEAD_REF master
    PATCHES
    some_patch.patch
)

vcpkg_list(SET OPTIONS)

vcpkg_list(APPEND OPTIONS "some_option")

vcpkg_configure_make(
    SOURCE_PATH "${SOURCE_PATH}"
    NO_WRAPPERS
    NO_ADDITIONAL_PATHS
    DETERMINE_BUILD_TRIPLET
    DISABLE_VERBOSE_FLAGS
    OPTIONS
        ${OPTIONS}
    OPTIONS_RELEASE
    OPTIONS_DEBUG
)

vcpkg_install_make()

file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")

vcpkg_fixup_pkgconfig()

file(INSTALL "${SOURCE_PATH}/license.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

**Comment by Neumann-A** [comment, marked as answer]
The helpers assume that you are correctly supporting `DESTDIR`. If you are not doing that than don't use `vcpkg_install_make`

The `_make` stuff will require a lot of rework in the future since it needs to split out its parts more fine grained.

**Reply by cpp977** [reply]
I will try to use `DESTDIR`. If I do not manage to use them, it's probably better to it all by hand.

**Reply by cpp977** [reply]
I could patch the external library to use `DESTDIR` and now all works as expected.

**Comment by dg0yt** [comment]
You should check the install log or the actual installed files in `/path/to/vcpkg/packages/<package_name>_<triplet>...`. IIRC a potential source of such symptoms is some or all commands not supporting the `DESTDIR` installation redirection.

**Reply by cpp977** [reply]
Okay, the files get currently directly installed into `path/to/build/vcpkg_installed/x64-linux` and not into the path `/path/to/vcpkg/packages/<package_name>_<triplet>...`. I will check the `DESTDIR` and how to support it. 
But, I am not setting any `--prefix` for the configure command this is set somewhere automatically in `vcpkg_configure_make`.
